bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/discovery/ElroConnectsBridgeDiscoveryService.java

```diff
@@ -0,0 +1,149 @@
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ElroConnectsBridgeDiscoveryService} is used to discover K1 Hubs attached to an ELRO Connects cloud account.
+ *
+ * @author Mark Herwege - Initial Contribution
+ */
+@NonNullByDefault
+public class ElroConnectsBridgeDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(ElroConnectsBridgeDiscoveryService.class);
+
+    private @Nullable ElroConnectsAccountHandler accountHandler;
+
+    private volatile @Nullable ScheduledFuture<?> discoveryJob;
+
+    private static final int TIMEOUT_S = 5;
+    private static final int REFRESH_INTERVAL = 60;
+
+    public ElroConnectsBridgeDiscoveryService() {
+        super(ElroConnectsBindingConstants.SUPPORTED_CONNECTOR_TYPES_UIDS, TIMEOUT_S);
+        logger.debug("Bridge discovery service started");
+        super.activate(null); // Makes sure the background discovery for devices is enabled
+    }
+
+    @Override
+    protected void startScan() {
+        discoverConnectors();
+    }
+
+    private void discoverConnectors() {
+        logger.debug("Starting hub discovery scan");
+        ElroConnectsAccountHandler account = accountHandler;
+        if (account == null) {
+            return;
+        }
+
+        ThingUID bridgeUID = account.getThing().getUID();
+
+        Map<String, ElroConnectsConnector> connectors = account.getDevices();
+        if (connectors == null) {
+            return;
+        }
+
+        connectors.entrySet().forEach(c -> {
+            if (c.getValue().online) {
+                String connectorId = c.getKey();
+                String firmwareVersion = c.getValue().binVersion;
+                boolean legacy = false;
+                try {
+                    legacy = !(Integer.valueOf(firmwareVersion.substring(firmwareVersion.lastIndexOf(".") + 1)) > 14);
+                } catch (NumberFormatException e) {
+                    // Assume new firmware if we cannot decode firmwareVersion
+                    logger.debug("Cannot get firmware version from {}, assume new firmware", firmwareVersion);
+                }
+                final Map<String, Object> properties = new HashMap<>();
+                properties.put(CONFIG_CONNECTOR_ID, connectorId);
+                properties.put(CONFIG_LEGACY_FIRMWARE, legacy);
+                properties.put("binVersion", c.getValue().binVersion);
+                properties.put("binType", c.getValue().binType);
+                properties.put("sdkVer", c.getValue().sdkVer);
+                properties.put("model", c.getValue().model);
```

Comment:
Here you can use standard property `Thing.PROPERTY_MODEL_ID`.

Reply:
Thanks.